### ISSUE TITLE
Azure service bus: authenticate with security token

### ISSUE THREAD

**Issue by aelmanaa** [Azure/azure-sdk-for-java#15722]
Hello,

in documentation I can only see how one can authenticate using a shared access signature. Is it possible to show how to authenticate using Azure AD acess token. In fact, we use service principals on our local development machines and managed identities on the cloud .  would be nice to use Azure AD authentication instead of sas

**Comment by joshfree** [comment]
Hi @aelmanaa, 
Our versioned [developer docs](https://azure.github.io/azure-sdk-for-java) has samples for how to authenticate using azure-identity and [azure-messaging-servicebus](https://azuresdkdocs.blob.core.windows.net/$web/java/azure-messaging-servicebus/7.0.0-beta.6/index.html) under the sections `Create a Service Bus client using Microsoft Identity platform (formerly Azure Active Directory)` and `Authorizing with DefaultAzureCredential`.

Please take a look and let us know if this helps get you unblocked; and if not, @hemanttanwar can help further in addition to updating the documentation with more samples/details.

Thank you.

**Comment by aelmanaa** [comment]
Hi @joshfree 

thanks for the info. I'm having another issue now:

com.azure.core.exception.ClientAuthenticationException: DefaultAzureCredential authentication failed. ---> VisualStudioCodeCredential authentication failed. Error Details: AADSTS65002: Consent between first party application 'xxx' and first party resource 'xxx' must be configured via preauthorization.

FYI: Locally I'm using intellij and i've setup the toolkit like explained here (https://docs.microsoft.com/en-us/azure/developer/java/toolkit-for-intellij/sign-in-instructions) and I've logged in using "Device Login"

**Comment by aelmanaa** [comment]
I've remarked that my IntelliJ stores everything in a hidden folder".AzureToolsForIntelliJ" . However it seems like the SDK looks for a folder which is not hidden "Paths.get(System.getProperty("user.home"), "AzureToolsForIntelliJ")" . Hence, I've copied ".AzureToolsForIntelliJ" into "AzureToolsForIntelliJ" . now I'm getting another issue : "The KeePass database path is either empty or not configured. Please configure it on the builder. It is required to use IntelliJ credential on the windows platform.
com.azure.identity.CredentialUnavailableException: The KeePass database path is either empty or not configured. Please configure it on the builder. It is required to use IntelliJ credential on the windows platform.
" Eventhough the path is properly setup in "C:\Users\User\AppData\Roaming\JetBrains\IdeaIC2020.2\c.kdbx"

**Comment by g2vinay** [comment]
> I've remarked that my IntelliJ stores everything in a hidden folder".AzureToolsForIntelliJ" . However it seems like the SDK looks for a folder which is not hidden "Paths.get(System.getProperty("user.home"), "AzureToolsForIntelliJ")" . Hence, I've copied ".AzureToolsForIntelliJ" into "AzureToolsForIntelliJ" . now I'm getting another issue : "The KeePass database path is either empty or not configured. Please configure it on the builder. It is required to use IntelliJ credential on the windows platform.
> com.azure.identity.CredentialUnavailableException: The KeePass database path is either empty or not configured. Please configure it on the builder. It is required to use IntelliJ credential on the windows platform.
> " Eventhough the path is properly setup in "C:\Users\User\AppData\Roaming\JetBrains\IdeaIC2020.2\c.kdbx"

Hi @aelmanaa 
That error indicates that you need to provide that path on the `IntelliJCredentialBuilder`
Here's the code sample:
```java
    IntelliJCredential intelliJCredential = new IntelliJCredentialBuilder()
        // KeePass configuration required only for Windows. No configuration needed for Linux / Mac
        .keePassDatabasePath("C:\\Users\\user\\AppData\\Roaming\\JetBrains\\IdeaIC2020.1\\c.kdbx")
        .build();
```

**Comment by aelmanaa** [comment]
HI @g2vinay 

thanks for your answer.  Actually I'm using DefaultAZureCredential instead of IntelliJCredentialBuilder. THe purpose being to use the chaining implementation which allows the same code to run locally (if a developer is using INtellij or VIsual studio code) or in the cloud (with Managed identity).
Probably I should close this thread and open a new one for azure-identity sdk. In fact, for windows users, even though the keepassdatabasePath is correctly set in Intellij settings, azure-identity sdk fails to fetch the user token

**Comment by g2vinay** [comment]
@aelmanaa 

For DefaultAzureCredential/IntelliJCredential,  on Windows, you will also need the KeePass database path to read IntelliJ credentials. You can find the path in IntelliJ settings under File -> Settings -> Appearance & Behavior -> System Settings -> Passwords. This path is not static and can be different for each dev environment.

![intellij keepass](https://github.com/Azure/azure-sdk-for-java/wiki/media/intellij-keepass.png)

Then once you figured out the path on your machine
you can specify the database path as follows:  
```java
        DefaultAzureCredential defaultAzureCredential = new DefaultAzureCredentialBuilder()
                         .intelliJKeePassDatabasePath("C:\\Users\\user\\AppData\\Roaming\\JetBrains\\IdeaIC2020.1\\c.kdbx")
                        .build();

```

If you still run into any authentication issue, please provide the error log and details with steps to reproduce and we can investigate it further.
